server/src/main/java/org/elasticsearch/index/mapper/TextFieldMapper.java

```diff
@@ -387,12 +391,15 @@ public Query prefixQuery(String value, MultiTermQuery.RewriteMethod method, Quer
             Automaton automaton = Operations.concatenate(automata);
             AutomatonQuery query = new AutomatonQuery(new Term(name(), value + "*"), automaton);
             query.setRewriteMethod(method);
-            return query;
+            return new BooleanQuery.Builder()
+                .add(query, BooleanClause.Occur.SHOULD)
```

Comment:
Is the `bool` query needed if `value.length` is greater than `minChars` ?

Reply:
If `value.length` is greater than or equal to `minChars` then we delegate to a term query, as before (see line 383).  This bit of the code is only used when `value.length` is less than `minChars`

Reply:
thanks for clarifying